src/script/calling/CallingRepository.ts

```diff
@@ -268,6 +272,22 @@ export class CallingRepository {
     }
   };
 
+  readonly updatePushToTalkKey = (key: string | null = null): void => {
+    this.unsubscribeFromPushToTalk();
+
+    if (key === null) {
+      return;
+    }
+
+    const activeCall = this.callState.joinedCall();
+
+    if (!activeCall) {
+      return;
+    }
+
+    this.subscribeToPushToTalk(key, activeCall);
+  };
+
```

Comment:
Hummm I was thinking this feature could live only in the UI layer and not have any impact on the CallingRepository. Is there anything needed in the calling repo at all? 

My idea was:

In the calling UI component, when mounted with an ongoing call => activate a react hook that would toggle the mic on/off
When unmounted unsubscribe. 

Would that be possible? 

I'd like to have the least amount of changes in the callingRepo